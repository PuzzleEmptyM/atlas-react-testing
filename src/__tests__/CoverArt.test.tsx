import { render } from '@testing-library/react';
import CoverArt from '../components/CoverArt';

describe('CoverArt Component', () => {
  it('renders with default props', () => {
    const { asFragment } = render(<CoverArt alt="Default cover art" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with specific image source', () => {
    const { asFragment } = render(
      <CoverArt src="test-image.jpg" alt="Test cover art" />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with different alt text', () => {
    const { asFragment } = render(
      <CoverArt alt="Alternative cover art" />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
