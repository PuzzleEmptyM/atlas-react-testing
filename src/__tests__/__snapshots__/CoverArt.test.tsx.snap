// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CoverArt Component > renders with default props 1`] = `
<DocumentFragment>
  <div
    class="w-image-w h-image-h overflow-hidden"
  >
    <img
      alt="Default cover art"
      class="w-full h-full object-cover"
      src="/src/assets/placeholder.svg"
    />
  </div>
</DocumentFragment>
`;

exports[`CoverArt Component > renders with different alt text 1`] = `
<DocumentFragment>
  <div
    class="w-image-w h-image-h overflow-hidden"
  >
    <img
      alt="Alternative cover art"
      class="w-full h-full object-cover"
      src="/src/assets/placeholder.svg"
    />
  </div>
</DocumentFragment>
`;

exports[`CoverArt Component > renders with specific image source 1`] = `
<DocumentFragment>
  <div
    class="w-image-w h-image-h overflow-hidden"
  >
    <img
      alt="Test cover art"
      class="w-full h-full object-cover"
      src="test-image.jpg"
    />
  </div>
</DocumentFragment>
`;
